share/spack/setup-tutorial-env.sh

```diff
@@ -0,0 +1,102 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+###############################################################################
+#
+# This file is part of Spack and sets up the environment for the Spack tutorial
+# It is intended to be run on ubuntu-18.04 or an ubuntu-18.04 container or AWS
+# cloud9 environment
+#
+# Components:
+# 1. apt installs for packages used in the tutorial
+#     these include compilers and externals used by the tutorial and
+#     basic spack requirements like python and curl
+# 2. spack configuration files
+#     these set the default configuration for Spack to use x86_64 and suppress
+#     certain gpg warnings. The gpg warnings are not relevant for the tutorial
+#     and the default x86_64 architecture allows us to run the same tutorial on
+#     any x86_64 architecture without needing new binary packages.
+# 3. aws cloud9 configuration to expand available storage
+#     when we run on aws cloud9 we have to expand the storage from 10G to 30G
+#     because we install too much software for a default cloud9 instance
+###############################################################################
+
+####
+# Install packages needed for tutorial
+####
+
+# compilers, basic system components, externals
+sudo apt-get update -y
+sudo apt-get install -y --no-install-recommends \
+    autoconf make python3 python3-pip \
+    build-essential ca-certificates curl git gnupg2 iproute2 emacs \
+    file openssh-server tcl unzip vim wget \
+    clang g++ g++-6 gcc gcc-6 gfortran gfortran-6 \
+    zlib1g zlib1g-dev mpich
+
+####
+# Spack configuration settings for tutorial
+####
+
+# create spack system config
+sudo mkdir -p /etc/spack
+
+# set default arch to x86_64
+sudo tee /etc/spack/packages.yaml << EOF > /dev/null
+packages:
+  all:
+    target: [x86_64]
+EOF
+
+# suppress gpg warnings
+sudo tee /etc/spack/config.yaml << EOF > /dev/null
+config:
+  suppress_gpg_warnings: true
+EOF
+
+####
+# AWS set volume size to 30G
+####
+
```

Comment:
can probably exit here if:

```bash
if [ ! -f /sys/hypervisor/uuid ]; then
    return
fi
```

This tells you it's an EC2 instance.  We can probably do better eventually but for now this is good.